### ISSUE TITLE
Remove `Main Stat assumption level`

### ISSUE THREAD

**Issue by frzyc** [frzyc/genshin-optimizer#2117]
The main stat assumption level is a feature that was added to allow users to include lower leveled artifacts in their optimization cycle. However, This feature has a lot of caveats:
- It needs to be fed into the calculation system, which causes a lot of wiring.
- It outputs an "incorrect calculation", since its using the "partial future value" of an artifact, and not the actual value of the artifacts in the build.
- Extra code logic needed to display the value in UI
- With implementation of Artifact Upgrader, its original function of finding under leveled artifacts is no longer powerful enough.

Due to the caveats above, I can no longer justify having this feature around.

**Comment by nguyentvan7** [comment]
I'm not really a fan of this. Artifact Upgrader still doesn't optimize a build, it just does a single swap. The use case of leveling up main stats is a separate use case from the artifact upgrader.

Also all the logic and wiring you are talking about is already built and established. It would be more work to remove it, than to keep it

**Comment by frzyc** [comment]
>Artifact Upgrader still doesn't optimize a build, it just does a single swap. The use case of leveling up main stats is a separate use case from the artifact upgrader.

I agree that it is not a direct trade for feature, but Upgrader does consider increases to main stat implicitly as well, and it does it better because it considers increases for substats. If a user is at a point to consider underleveled artifacts, i doubt the "single swap" limitation of Upgrader would be holding them back. 

> Also all the logic and wiring you are talking about is already built and established. It would be more work to remove it, than to keep it

I would incline to agree with you on this part, however, this feature has presented some issues on some of the future changes I want to make, most recently around standardizing build compare UI (since it shows the wrong value due to inflated main stat value).
